### PR TITLE
Update wine{unstable,staging,mono,winetricks}

### PR DESCRIPTION
**Pull request by nyanloutre**
###### Motivation for this change

* wine{unstable,staging}: 4.2 -> 4.5
* wine-mono: 4.7.5 -> 4.8.0
* winetricks: 20181203 -> 20190310

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by nyanloutre** [comment]
I tested wineWowPackages.unstable successfully

File: pkgs/misc/emulators/wine/sources.nix
Comment (tadeokondrak) [inline pkgs/misc/emulators/wine/sources.nix]:
```suggestion
    version = "4.5";
```

File: pkgs/misc/emulators/wine/sources.nix
Comment (tadeokondrak) [inline pkgs/misc/emulators/wine/sources.nix]:
```suggestion
    sha256 = "1dy1v27cw9vp2xnr8y4bdcvvw5ivcgpk2375jgn536csbwaxgwjz";
```

File: pkgs/misc/emulators/wine/sources.nix
Comment (tadeokondrak) [inline pkgs/misc/emulators/wine/sources.nix]:
```suggestion
    sha256 = "18xpha7nl3jg7c24cgbncciyyqqb6svsyfp1xk81993wnl6r8abs";
```

**Comment by nyanloutre** [comment]
If you want to test wineWowPackages.unstable I pushed it to cachix : https://nyanloutre.cachix.org/

**Comment by ikervagyok** [comment]
why does it always take such a long time until wine updates get merged? wine 4.6 is already out (also staging). anyway - would you update and then we find someone to finally commit these changes?

**Comment by veprbl** [comment]
@GrahamcOfBorg build wine winePackages.staging winePackages.full

**Comment by veprbl** [comment]
cc maintainers @avnik @7c6f434c @bendlas @the-kenny

**Comment by nyanloutre** [comment]
@veprbl the borg build usually timeout because wine is taking more than an hour to build

**Review by veprbl — APPROVED** [review]

**Comment by veprbl** [comment]
Passes nix-review, winemine works.

**Comment by nyanloutre** [comment]
@ikervagyok I made a PR for 4.6 here : #59764

**Pull request merged**
